src/Pattern.re

```diff
@@ -0,0 +1,43 @@
+module Tile = {
+  type t = array(array(int));
+
+  let from_bytes = (bytes: bytes, offset: int) => {
+    Array.init(8, y =>
+      Array.init(
+        8,
+        x => {
+          let low = Bytes.get(bytes, 16 * offset + y) |> Char.code;
+          let high = Bytes.get(bytes, 16 * offset + y + 8) |> Char.code;
+
+          (Util.read_bit(low, 7 - x) ? 1 : 0)
+          + (Util.read_bit(high, 7 - x) ? 2 : 0);
+        },
+      )
+    );
+  };
+
+  let from_codes = (cs): t =>
+    from_bytes(Bytes.init(16, i => Char.chr(cs[i])), 0);
+
+  let inspect = (tile: t, format: int => string): string => {
+    let result = ref("");
+
+    Array.iter(
+      row => {
+        Array.iter(n => result := result^ ++ format(n), row);
+        result := result^ ++ "\n";
+      },
+      tile,
+    );
+
+    result^;
+  };
+};
+
+module Table = {
+  type t = array(Tile.t);
+
+  let load = (rom: Rom.t): array(Tile.t) => {
+    Array.init(256, i => Tile.from_bytes(rom.chr, i));
```

Comment:
I'm not wild about this / sure that it is correct. I feel like we should be doing something closer to:

```
Array.init(rom.chr_count * 512, i => Tile.from_bytes(rom.chr, i));
```

What you have works but most Sprite banks are divided into two 256 tile sections: one for backgrounds and one for other graphics. Also, this will (probably?) error on any games that intermingle code and graphics and have an empty CHR section. We won't really hit them until we start handling non-NROM games but it seems a small enough tweak to be worth doing now.

Reply:
Realized I should explain my math above. I have 512 above since CHR data occurs in cartridges in 8k blocks (from the mappers reference, no 4k CHR games, and `512 * 16` is 8k). The nes rom file format (and `Rom.re`) count in 8k chunks for `chr_count` and `chr_size` which is a bit annoying seeing as some mappers (MMC1 and MMC3, at least) allow switching CHR banks in 4k chunks. 🤷‍♂️

Reply:
Agreed. This is one that I thrashed around on trying to get something "right" and then decided it was better to just get _something_; I feel like we can't write the correct thing here without tackling better mapper support first.

I do think your suggestion is a definite improvement as it stands though, so I'm going to apply it.